core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -872,6 +917,423 @@ export class ExplorationEditor extends BaseUser {
+        filteredElements.map(async el => {
+          return el.evaluate(el => {
+            // The value of the property is held in the third sibling span element.
+            if (
+              el.nextElementSibling &&
+              el.nextElementSibling.nextElementSibling
+            ) {
+              return el.nextElementSibling.nextElementSibling.textContent;
+            } else {
+              throw new Error(`Unable to find "${property}" property's value`);
+            }
+          });
+        })
+      ).then(results => results.filter(result => result !== null) as string[]);
+    }
+
+    for (let property in values) {
+      if (values[property][0] === values[property][1]) {
+        throw new Error(`No changes are reflected in ${property}`);
+      }
+      showMessage(`Changes are reflected in ${property}.`);
+    }
+  }
+
+  /**
+   * Verifies if metadata changes are reflected for each property within a single code block.
+   * @param {string[]} properties - The properties to check for.
+   */
+  async expectMetadataChangesInProperties(properties: string[]): Promise<void> {
+    await this.page.waitForSelector(viewMetadataChangesButton, {visible: true});
+    await this.clickOn(viewMetadataChangesButton);
+
+    await this.expectChangesInProperties(properties);
+
+    await this.clickOn(closeMetadataModal);
+    await this.page.waitForSelector(closeMetadataModal, {hidden: true});
+  }
+
+  /**
+   * Verifies if exploration state changes are reflected for each property within a single code block.
+   * @param {string[]} properties - The properties to check for.
+   */
+  async expectStateChangesInProperties(properties: string[]): Promise<void> {
+    // Add two spaces in front of each property to match the formatting in the UI.
+    const formattedProperties = properties.map(property => `  ${property}`);
+
+    await this.clickOn(testNodeBackground);
+    await this.expectChangesInProperties(formattedProperties);
+
+    await this.clickOn(closeStateModal);
+  }
+
+  /**
+   * Resets the comparison results by clicking on the reset graph button.
+   * @returns {Promise<void>}
+   */
+  async resetComparisonResults(): Promise<void> {
+    await this.page.waitForSelector(resetGraphButton);
+    await this.clickOn(resetGraphButton);
+  }
+
+  /**
+   * Finds the options button for a specific version.
+   * @param {number} version - The version number to find.
+   */
+  private async findOptionsButtonForVersion(
+    version: number
+  ): Promise<ElementHandle> {
+    const buttons = await this.page.$$('.history-table-option');
+    const index = buttons.length - version;
+    if (index < 0 || index >= buttons.length) {
+      throw new Error(`No button found for version ${version}`);
+    }
+    return buttons[index];
+  }
+
+  /**
+   * Downloads a specific version.
+   * @param {number} version - The version number to be downloaded.
+   */
+  async downloadVersion(version: number): Promise<void> {
```

Comment:
@imchristie @seanlip @StephenYu2018 
I need to confirm this download, wherein this download initiates on clicking the download button which in turn opens a new tab. However, this tab closes too quickly to read any event or fetch its URL for download confirmation. I attempted to monitor network activity, but the file appears to be served directly, without any network request. Additionally, I'm unable to access the download directory as this is set to run on CI. 
I tried looking if puppeteer provides some way to confirm the download, but couldn't find any.
May you please share some approach to confirm the download of the revisions? 

Thank you.

Reply:
@Akhilesh-max I don't know offhand, but what online resources have you looked at for handling downloads in puppeteer? That's where I would start -- a quick search showed a bunch of results.